lib/src/model/notifications/challenge_notification.dart

```diff
@@ -0,0 +1,226 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:lichess_mobile/l10n/l10n.dart';
+import 'package:lichess_mobile/l10n/l10n_en.dart';
+import 'package:lichess_mobile/src/model/challenge/challenge.dart';
+import 'package:lichess_mobile/src/model/challenge/challenge_repository.dart';
+import 'package:lichess_mobile/src/model/challenge/challenges.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/model/notifications/local_notification_service.dart';
+import 'package:lichess_mobile/src/navigation.dart';
+import 'package:lichess_mobile/src/utils/l10n.dart';
+import 'package:lichess_mobile/src/utils/navigation.dart';
+import 'package:lichess_mobile/src/view/game/game_screen.dart';
+import 'package:lichess_mobile/src/view/user/challenge_requests_screen.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'challenge_notification.g.dart';
+
+@Riverpod(keepAlive: true)
+ChallengeService challengeService(ChallengeServiceRef ref) {
+  return ChallengeService(ref);
+}
+
+class ChallengeService {
+  ChallengeService(this.ref);
+
+  final ChallengeServiceRef ref;
+
+  void init() {
+    ref.listen(challengesProvider, (prev, current) {
```

Comment:
Here you should listen `socketGlobalStream`, the server holds all the state and is the source of truth. Also we don't want the service to depend on `challengesProvider`.

If you want to avoid parsing twice the socket json object, I'd do it the other way around: export a `Stream<ChallengesList>` in the `ChallengeService` (this one is kept alive because it is not bound to a view, and so we can access it anywhere), that you'd consume in the `challengesProvider`.